src/stream/src/executor/source/fs_source_executor.rs

```diff
@@ -104,11 +105,13 @@ impl<S: StateStore> FsSourceExecutor<S> {
             source_desc.source.config.clone(),
             self.stream_source_core.source_name.clone(),
         );
-        source_desc
+        let stream = source_desc
             .source
             .to_stream(state, column_ids, Arc::new(source_ctx))
             .await
-            .map_err(StreamExecutorError::connector_error)
+            .map_err(StreamExecutorError::connector_error)?;
+
+        Ok(apply_rate_limit(stream, self.source_ctrl_opts.rate_limit).boxed())
```

Comment:
Are you sure the `apply_rate_limit` will be called whenever `source_ctrl_opts.rate_limit` was changed?

Reply:
No for fs source and arrangement backfill. They do not handle throttle mutation. 
I am going to implement this part for fs source in this pr. For arrangement backfill, it relies on restarting to apply the change @kwannoel

Reply:
Add alter for arrangement backfill via barrier mutation after this PR.

Reply:
Please make sure to complete it before releasing the function to users. 🙏

Reply:
Fixed